SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -319,7 +317,15 @@
       <!-- Check if the ruleset exists -->
       <SonarQubeRulesetFullName>$(SonarQubeConfigPath)\$(SonarQubeRulesetFileName)</SonarQubeRulesetFullName>
       <SonarQubeRulesetExists>$([System.IO.File]::Exists($(SonarQubeRulesetFullName)))</SonarQubeRulesetExists>
+    </PropertyGroup>
+  </Target>
 
+  <!-- We want to override any properties that have been set declaratively in the project -->
+  <Target Name="OverrideCodeAnalysisProperties" Condition=" $(SonarQubeTempPath) != '' "
+          DependsOnTargets="SonarQubeCategoriseProject"
+          BeforeTargets="RunCodeAnalysis" >
```

Comment:
This target now depends on *DetectFxCopRuleset* (to set *SonarQubeRulesetExists*). The target should explicitly depend on *DetectFxCopRuleset*, and the existing FxCop tests should include *DetectFxCopResult* in the target-order checks.